src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs

```diff
@@ -168,33 +174,50 @@ internal ManifestItem Transform(InternalManifestItem item)
                     var message = $"Error processing {item.LocalPathFromRoot}: {e.Message}";
                     throw new PathTooLongException(message, e);
                 }
+            }
+
+            LogInvalidXRefExceptions(aes);
+
+            return manifestItem;
+        }
 
+        private void LogInvalidXRefExceptions(List<AggregateException> aes)
+        {
+            if (aes == null || aes.Count == 0)
+            {
+                return;
             }
 
-            if (missingXRefDetails.Count > 0)
+            var invalidXRefs = (from ae in aes
+                                      from ie in ae.InnerExceptions
+                                      select ie as InvalidCrossReferenceException into ce
+                                      where ce != null
+                                      select ce.XRefDetails).Distinct().ToList();
+
+            if (invalidXRefs.Count == 0)
             {
-                var distinctUids = string.Join(", ", missingXRefDetails.Select(i => i.RawSource).Distinct().Select(s => $"\"{s}\""));
-                Logger.LogWarning($"Invalid cross reference {distinctUids}.", file: item.LocalPathFromRoot);
-                foreach (var group in missingXRefDetails.GroupBy(i => i.SourceFile))
-                {
-                    int maxCountPerFile = 10;
-                    string details;
+                return;
+            }
 
-                    // For each source file, print the first 10 invalid cross reference
-                    if (group.Count() >= maxCountPerFile)
-                    {
-                        details = $"{string.Join(", ", group.Take(maxCountPerFile).Select(i => $"\"{i.RawSource}\" in line {i.SourceStartLineNumber.ToString()}"))}, etc. Fisrt {maxCountPerFile} invalid cross reference are shown above.";
-                    }
-                    else
-                    {
-                        details = string.Join(", ", group.Select(i => $"\"{i.RawSource}\" in line {i.SourceStartLineNumber.ToString()}"));
-                    }
+            var distinctUids = invalidXRefs.Select(i => i.RawSource).Distinct().Select(s => $"\"{HttpUtility.HtmlDecode(s)}\"").ToList();
```

Comment:
Nice catch, synced with @fenxu , will integrate with them when our change is in.